File: 228-summary-ranges/solution.ts
function summaryRanges(nums: number[]): string[] {
  let start: number | null = null;
  let answer: string[] = [];
  for (let i = 0; i <= nums.length; i++) {
    if (start === null) start = nums[i];
    else if (nums[i] !== nums[i - 1] + 1) {
      if (start < nums[i - 1]) answer.push(`${start}->${nums[i - 1]}`);
      else answer.push(`${start}`);
      start = nums[i];
    }
  }
  return answer;
}

File: 228-summary-ranges/solution.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if(nums.size() == 0) return {};
        if(nums.size() == 1) return {to_string(nums[0])};
        vector<string> ans;
        int left = nums[0];
        for(int i = 1; i <= nums.size(); i++) {
            if(i == nums.size() || nums[i] != nums[i - 1] + 1) {
                int right = nums[i - 1];
                ans.push_back(left == right ? to_string(left) : format("{}->{}", left, right));
                if(i < nums.size()) left = nums[i];
            }
        }
        return ans;
    }
};
